db/helpers/generatorUtil.js

```diff
@@ -13,6 +13,39 @@ const common = require('./common');
 const dbErrors = require('../dbErrors');
 
 /**
+ * Function to validate the context field of the sample generator based on the
+ * contextDefinition field of its related generator template.
+ * @param  {Object} sgCtx - The generator context object     [description]
+ * @param  {Object} sgtCtxDef - The related sample generator template context
+ * definition object
+ * @throws {MissingRequiredFieldrror} If the generator context field does not
+ * have the attributes that are required by the context definiton field of the
+ * sample generator template
+ * @throws {ValidationError} If the number of keys in the generator context do
+ * not match the number of keys in the generator template context.
+ */
+function validateGeneratorCtx(sgCtx, sgtCtxDef) {
+  const sgtCtxDefKeys = sgtCtxDef ? Object.keys(sgtCtxDef) : [];
+  const sgCtxKeys = sgCtx ? Object.keys(sgCtx) : [];
+
+  if (sgCtxKeys.length > sgtCtxDefKeys.length) {
```

Comment:
I think you'd also want to do something like this:

```
const invalidKeys = sgCtxKeys.filter((key) => !sgtCtxDefKeys.includes(key));
if (invalidKeys.length) {
  throw new dbErrors.ValidationError(
  { explanation: `Sample generator context contains invalid keys: ${invalidKeys}` });
}
```

Reply:
Nice. I think just this will handle both the cases. I will add this.